app/views/Main.js

```diff
@@ -7,82 +7,78 @@ import Colors from '../constants/colors';
 import { isGPS } from '../COVIDSafePathsConfig';
 import { checkIntersect } from '../helpers/Intersect';
 import BackgroundTaskServices from '../services/BackgroundTaskService';
-import LocationServices, { Reason } from '../services/LocationService';
-import { ExposureNotificationNotAvailablePage } from './main/ExposureNotificationNotAvailablePage';
+import LocationServices from '../services/LocationService';
+import ExposureNotificationService from '../services/ExposureNotificationService';
 import { ExposurePage } from './main/ExposurePage';
 import { NoKnownExposure } from './main/NoKnownExposure';
 import { OffPage } from './main/OffPage';
 import { styles } from './main/style';
 import { UnknownPage } from './main/UnknownPage';
 
 export const Main = () => {
+  const tracingService = isGPS ? LocationServices : ExposureNotificationService;
   const navigation = useNavigation();
   if (isPlatformAndroid()) {
     StatusBar.setBackgroundColor(Colors.TRANSPARENT);
     StatusBar.setBarStyle('light-content');
     StatusBar.setTranslucent(true);
   }
 
-  const [location, setLocation] = useState({
+  const [trackingInfo, setTrackingInfo] = useState({
     canTrack: true,
     reason: null,
     hasPotentialExposure: false,
   });
 
   const checkForPossibleExposure = () => {
-    BackgroundTaskServices.start();
-    checkIntersect();
+    if (isGPS) {
+      BackgroundTaskServices.start();
+      checkIntersect();
+    }
   };
 
   const updateStateInfo = useCallback(async () => {
     checkForPossibleExposure();
-    const state = await LocationServices.checkStatusAndStartOrStop();
-    setLocation(state);
-  }, [setLocation]);
+    const state = await tracingService.checkStatusAndStartOrStop();
+    setTrackingInfo(state);
+  }, [tracingService, setTrackingInfo]);
 
   const handleBackPress = () => {
     BackHandler.exitApp(); // works best when the goBack is async
     return true;
   };
 
   useEffect(() => {
-    if (isGPS) {
```

Comment:
should this be removed?

Reply:
If we use the ExposureNotificationService, I think we can have the same implementation. Meaning that the ExposureNotificationService would be called to check and update the stateInfo similarly to GPS.